File: String/Implement Atoi/atoi.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  int atoi(string str) {
    int neg = 1;
    int i = 0;
    if (str[0] == '-') {
      i++;
      neg = -1;
    }
    int ans = 0;
    for (i; i < str.size(); i++) {
      if (str[i] <= '9' and str[i] >= '0')
        ans = ans * 10 + str[i] - 48;
      else
        return -1;
    }
    return ans * neg;
  }
};

int main() {
  int t;
  cin >> t;
  while (t--) {
    string s;
    cin >> s;
    Solution ob;
    int ans = ob.atoi(s);
    cout << ans << endl;
  }
}

File: String/Implement Atoi/Atoi.java
import java.util.Scanner;

class Atoi {
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    sc.nextLine();
    while (t > 0) {
      String str = sc.nextLine();
      Solution obj = new Solution();
      int num = obj.atoi(str);
      System.out.println(num);
      t--;
    }
    sc.close();
  }
}

class Solution {
  int atoi(String s) {
    int neg = 1;
    int i = 0;
    if (s.charAt(0) == '-') {
      i++;
      neg = -1;
    }
    int ans = 0;
    for (; i < s.length(); i++) {
      char c = s.charAt(i);
      if (c >= '0' && c <= '9') {
        ans = ans * 10 + (c - '0');
      } else {
        return -1;
      }
    }
    return ans * neg;
  }
}
